### PR TITLE
Code cleanups

### PR DESCRIPTION
**Pull request by rouault**
- QgsAttributeTableDialog::mActionToggleEditing_toggled(): reduce amount of copy&paste related to provider capabilities
- Reduce number of calls to virtual method temporalCapabilities()

### CODE REVIEW

File: src/providers/arcgisrest/qgsafsprovider.cpp [hunk 1]
```diff
@@ -249,20 +249,21 @@ QgsAfsProvider::QgsAfsProvider( const QString &uri, const ProviderOptions &optio
...
     const QVariantMap timeInfo = layerData.value( QStringLiteral( "timeInfo" ) ).toMap();
 
-    temporalCapabilities()->setHasTemporalCapabilities( true );
-    temporalCapabilities()->setStartField( timeInfo.value( QStringLiteral( "startTimeField" ) ).toString() );
-    temporalCapabilities()->setEndField( timeInfo.value( QStringLiteral( "endTimeField" ) ).toString() );
-    if ( !temporalCapabilities()->endField().isEmpty() )
-      temporalCapabilities()->setMode( Qgis::VectorDataProviderTemporalMode::StoresFeatureDateTimeStartAndEndInSeparateFields );
-    else if ( !temporalCapabilities()->startField().isEmpty() )
-      temporalCapabilities()->setMode( Qgis::VectorDataProviderTemporalMode::StoresFeatureDateTimeInstantInField );
+    QgsVectorDataProviderTemporalCapabilities *l_temporalCapabilities = temporalCapabilities();
```
Comment (nyalldawson) [inline src/providers/arcgisrest/qgsafsprovider.cpp:252]: I've seen this elsewhere but never asked -- what's the "l_" prefix signify? Does it stand for something?

Comment (rouault) [reply src/providers/arcgisrest/qgsafsprovider.cpp:252]: l_ = local variable. As a way to disambiguate the virtual method foo() and the local variable storing its value.
But actually I see in src/core/qgsmaplayer.cpp a "const QgsDataProvider *lDataProvider = dataProvider()". Is "lFoo" a clearer convention than "l_foo" ?

Comment (nyalldawson) [reply src/providers/arcgisrest/qgsafsprovider.cpp:252]: I think so -- _ isn't used widely in qgis naming.

Comment (rouault) [reply src/providers/arcgisrest/qgsafsprovider.cpp]: ok, commit ammended to use lFoo convention

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**